mod/photos.php

```diff
@@ -914,7 +914,47 @@ function photos_content(App $a)
 			'$submit' => DI::l10n()->t('Submit'),
 		]);
 
-		$usage_message = '';
+		/* This block determines which setting actually limits  upload size of images:
+		 * limit of Friendica ('maximagesize') or upload_max_filesize
+		 * and outputs the lower one.
+		*/
+		// init output var in case no number value could be retrieved.
+		$maximagesize_Mbytes = 0;
+		$sizelimitedby = '';
+		{
+			// Get the relevant size limits for uploads. Abbreviated var names: MaxImageSize -> mis; upload_max_filesize -> umf
+			$mis_bytes = DI::config()->get('system', 'maximagesize');
+			// get_cfg_var('upload_max_filesize') outputs a string in the shorthand notation, need to convert it to bytes
+			$umf_string = get_cfg_var('upload_max_filesize');
```

Comment:
Please use `ini_get()` instead, it will return the computed value as an integer value or an integer string you can use directly.

Reply:
At least not on my setup

> `gettype(ini_get('upload_max_filesize')): String`
> `ini_get('upload_max_filesize'): 20M`

Reply:
Gasp!

Never mind then, here's a conversion function you should use (notice the 1024 factor instead of the 1000 you've been using): https://stackoverflow.com/a/6846537

You can declare it in the `src/Util/Strings.php` file with the signature `public static function getBytesFromShorthand(string $val): int` and then you can refer to it in `mod/photos.php` as `Strings::getBytesFromShorthand()`.

Reply:
Thanks, I will do so shortly.

Already thought about the binary / si prefix issue myself, which is also related to the [issue](https://github.com/friendica/friendica/issues/7807): Whenever I encounter a si prefix (kB, MB, ...) I assume it is decimal prefix `10**x` . 
When it is about binary prefixes, shouldn't we use KiB, MiB, Gib? 

But for the moment: Which interpretation of k, M, G is used in this project: `10**x`  (si / deicmal) or `2**x` (binary). I never be sure in the k, M, G cases, only when binary prexies are used it is celar for me that they relate to `2**x`  :-)

Reply:
The PHP manual you linked to earlier mentions the shorthands are meant to be binary prefixes.

Reply:
> Never mind then, here's a conversion function you should use (notice the 1024 factor instead of the 1000 you've been using): https://stackoverflow.com/a/6846537

I guess the code snippet there is not perfectly correct, because:

```
> switch($last) {
        // The 'G' modifier is available since PHP 5.1.0
        case 'g':
            $val *= 1024;
        case 'm':
            $val *= 1024;
        case 'k':
            $val *= 1024;
    }
```
will basically always just multiply the number in from the shorthand notation by 1024 and thus only lowering it by one power (e.g. convert to MiB to KiB). I will adjust it to:
```
  switch($last) {
        // The 'G' modifier is available since PHP 5.1.0
        case 'g':
            $val *= 1024**3;
        case 'm':
            $val *= 1024**2;
        case 'k':
            $val *= 1024;
    }
```
I am confused and unsettled that nobody noticed over there at stackoverflow, or is there a big misconception on my side?

Reply:
Ah, no fault, the case environment just does not use break statements so that it is "powered down" step by step. I was wrong, already wondered :-)
My previous comment can be therefore ignored.